src/components/scan/Scan.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { StyleSheet, View, ViewStyle } from 'react-native';
+import { useTheme } from '../../styles/themes';
+import { Icon } from '../icons/Icon';
+import { Body } from '../typography/Body';
+export type ScanState = 'default' | 'selected' | 'filled';
+interface Props {
+    text: string;
+    state: ScanState;
+    style?: ViewStyle;
+}
+
+export const Scan = ({ text, state, style }: Props) => {
```

Comment:
I ask to design team about the name. I find this name unclear whereas ScanInput seems more explicit about the component behavior.

Reply:
Alright, i will rename it